timemachine/integrator.py

```diff
@@ -66,7 +71,55 @@ def multiple_steps(self, x, v, n_steps=1000):
         return np.array(xs), np.array(vs)
 
 
-class LangevinIntegrator(Integrator):
+class StochasticIntegrator(ABC):
+    @abstractmethod
+    def step(self, x, v, rng) -> Tuple[Any, Any]:
```

Comment:
I think this looks good!

Comment (not a change request): Another alternative might be to have `rng` be an instance attribute referenced by `step` implementation (`self.rng.randn(...)`), rather than an explicit argument of the `step` function. (Cons: This wouldn't make `step` deterministic, and it wouldn't be consistent with the jax rng approach. Pros: It would avoid dependence on global rng state, and it would be [similar to the C++ interface](https://github.com/proteneer/timemachine/blob/5aa164035823e13d58aff2c01ad2d71223c25dd0/timemachine/cpp/src/integrator.hpp#L26-L29), where `rng` is a private class member and `seed` can only be set in constructor.)

Reply:
> Another alternative might be to have rng be an instance attribute referenced by step implementation

I think @badisa suggested this too (https://github.com/proteneer/timemachine/pull/786#discussion_r935629249). My inclination is to avoid having the RNG as an instance variable, since it seems to reduce flexibility a bit (https://github.com/proteneer/timemachine/pull/786#discussion_r935944325) relative to allowing RNG instances to be passed to member functions. It does seem preferable to global RNG state though (but I'd be in favor of adopting something like the more explicit style encouraged by jax).